fix: Fixed some rendering behaviors of box-shadow

lizidev opened the pull request:
* Fix shadow clipping.
* Fixed shadow rendering order.
* Support for the spread property of shadows.

lizidev commented:
`error: this if statement can be collapsed` error in cargo clippy, I don't know which is the better solution:

* Add `rust-version` to all crates, [related issues](https://github.com/rust-lang/rust-clippy/issues/14850). 
* Add `#[allow(clippy::collapsible_if)]` to all errors.

nicoburns commented:
@lizidev
- I've updated CI to use stable rust rather than nightly rust for clippy.
- I've updated `color` on `main` so you shouldn't need that here
- Do you have screenshots, tests, or sample code which demonstrates the rendering features you have fixed here?

lizidev commented:
Using the example I added, the order of the list below corresponds to the order in the screenshots:

* Fix shadow clipping.
* Fixed shadow rendering order.
* Support for the spread property of shadows.

| main branch |  This PR |
|-| -|
|![2561747752446_ pic](https://github.com/user-attachments/assets/18963ce6-90dc-446a-96b7-0e030d14a4d9)|![2571747752542_ pic](https://github.com/user-attachments/assets/1b6ca833-01b8-4649-b54e-8834975d93a9)|

nicoburns commented:
This seems to fix the shadow rendering for the `todomvc` example - nice!
(see: the "stacked boxes" effect at bottom of the central card)

<img width="912" alt="Screenshot 2025-05-20 at 17 01 00" src="https://github.com/user-attachments/assets/4ba583b4-67a4-460b-89bb-ca5d0ac0d830" />
